rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/editProjectFileService.ts

```diff
@@ -0,0 +1,69 @@
+import {getRundeckContext} from "../../../library";
+import {Notification} from "uiv";
+
+const rundeckClient = getRundeckContext().rundeckClient
+function notifySuccess(title: string, message: string) {
+    Notification.notify({
+        type: "success",
+        title: title,
+        content: message,
+        duration: 5000
+    });
+}
+
+function notifyError(message: string) {
+    Notification.notify({
+        type: "danger",
+        title: "An Error Occurred",
+        content: message,
+        duration: 0
+    });
+}
+
+function notifyWarning(message: string) {
+    Notification.notify({
+        type: "info",
+        title: "Warning",
+        content: message,
+        duration: 0
+    });
+}
+export async function saveProjectFile(project: string, filename: string, fileText: string) {
+    const resp = await rundeckClient.sendRequest({
+        baseUrl: `${getRundeckContext().rdBase}api/${getRundeckContext().apiVersion}`,
+        pathTemplate: "/project/" + project + "/" + filename,
+        method: "PUT",
+        body: {
+            contents: fileText
+        }
+    });
+
+    if (resp.status === 200) {
+        notifySuccess("Success", "Saved Project File " + filename);
+    } else {
+        notifyError(resp.parsedBody.message);
+    }
+}
+export async function getFileText(project: string, filename: string) {
+    try {
```

Comment:
would tweak this code to something more similar to this:

```
export async function getFileText () {
  return rundeckClient.sendRequest([data here]).then(response => {
        if(response.status === 200){

          return [response data or parsedBody]
        }
        else{
          // handle the error logic here, but instead of calling notifyWarning or notifyError, 
          // throwing an object that has the message and any other properties that might be helpful
          // for the component to decide how to handle this error
        }
}
```